vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/screenshot/ScreenCapturerTestRunListener.kt

```diff
@@ -66,17 +68,16 @@ class ScreenCapturerTestRunListener(
     override suspend fun testEnded(test: TestIdentifier, testMetrics: Map<String, String>) {
         super.testEnded(test, testMetrics)
         val toTest = test.toTest()
+        lastTestIdentifier = null
         logger.debug { "Finished recording for ${toTest.toSimpleSafeTestName()}" }
         supervisorJob?.cancelAndJoin()
         if (!hasFailed && screenRecordingPolicy == ScreenRecordingPolicy.ON_FAILURE) {
             supervisorJob?.invokeOnCompletion {
```

Comment:
Is `invokeOnCompletion` called when `job` is canceled?

Reply:
yes, precisely